src/sentry/ingest/consumer/processors.py

```diff
@@ -53,10 +53,32 @@ def inner(*args, **kwargs):
     return wrapper
 
 
+def process_transaction_no_celery(
+    data: MutableMapping[str, Any], project_id: int, cache_key: str, start_time: float
+) -> None:
+    manager = EventManager(data)
```

Comment:
> I omitted the metrics since we will get the kafka ones, which are aligned with the other pipelines

now there are two ways to measure transactions latency. i suggest to defer these changes for when you know that inlining the celery task will work.

> do we actually need the sdk scopes? The set_current_event_project looks like it's only relevant for celery

yes, that stuff is just basic instrumentation for when a crash happens

Reply:
> I'd prefer not to port stuff unless absolutely needed

Metrics and killswitches are a must. 
If you want to change the way to record some of those metrics, please see where they are used in sandboxes and alerts.

Reply:
i added back the killswitch and the `set_current_event_project` thing. I did not put back the metric. The queues dashboard does not apply anymore since there is no celery queue involved, so i don't think it makes sense to try and capture this there. The ingest-transactions consumer already has metrics and monitors attached to it, and the e2e latency measured by Snuba still applies same as before.

Reply:
Have you looked around for who uses that metric? The queue dashboard was only an example.